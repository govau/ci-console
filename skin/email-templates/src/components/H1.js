import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const StyledH1 = styled.h1`
  font-size: 24px;
  text-align: ${({ align }) => align};
`;

const H1 = ({ children, align }) => (
  <StyledH1 align={align}>{children}</StyledH1>
);

H1.propTypes = {
  align: PropTypes.oneOf(["left", "right", "center"]).isRequired
};

H1.defaultProps = {
  align: "left"
};

export default H1;
