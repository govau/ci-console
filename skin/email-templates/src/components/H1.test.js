import React from "react";
import renderer from "react-test-renderer";

import H1 from "./H1";

test("renders without crashing", () => {
  const tree = renderer.create(<H1>content</H1>);
  expect(tree).toMatchSnapshot();
});
