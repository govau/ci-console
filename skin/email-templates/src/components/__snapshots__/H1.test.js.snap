// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
.c0 {
  font-size: 24px;
  text-align: left;
}

<h1
  className="c0"
>
  content
</h1>
`;
